ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1308,8 +1399,13 @@ GetOption getGetOption() {
      */
     @Override
     MessageFormatFlags getOperationFlag() {
-      return options.getBlobOptions.getOperationType() == GetBlobOptions.OperationType.Data ? MessageFormatFlags.Blob
-          : MessageFormatFlags.All;
+      if (isMetadataCacheHit()) {
+        return MessageFormatFlags.BlobInfo;
+      } else if (options.getBlobOptions.getOperationType() == GetBlobOptions.OperationType.Data) {
+        return MessageFormatFlags.Blob;
+      } else {
+        return MessageFormatFlags.All;
+      }
```

Comment:
this operationFlag is for requests to server node, it shouldn't have anything to do with cache.